File: swig/perl/desc.pl
#! /usr/bin/perl -w
#
use greendb;
use strict;

sub desc {
	my $env = greendb::new_GreenEnv(".");
	greendb::GreenEnv_open($env);
  my $table_name = $ARGV[0];
  print "creating " . $table_name . "\n";
	my $table = greendb::new_Table($table_name, $env);
	my $schema = greendb::Table_get_schema($table);

  my $size = greendb::Schema_size($schema);
  for my $i (0 .. ($size -1) ) {
    my $type = greendb::Schema_get_type_n($schema, $i);
    my $name = greendb::Schema_get_name($schema, $i);
    print $i . " " . $type . " " . $name . "\n";
  }
	greendb::Table_close($table);
	greendb::GreenEnv_close($env);
}

desc();

File: swig/python/desc.py
import greendb
import sys

def desc(table_name):
	env = greendb.GreenEnv(".")
	env.open()
	
	table = greendb.Table("testtable", env)
	schema = table.get_schema()
	print table.get_name()	
	i = 0
	for i in range(0, schema.size()):
		type = schema.get_type_n(i)
		name = schema.get_name(i)
		print i, type, name
	
	table.close()
	env.close()

if __name__ == '__main__':
	desc(sys.argv[1])
